### DISCUSSION TITLE
Gitbook embed feature

### DISCUSSION THREAD

**Discussion by sheiquezakir** [GitbookIO/community#504] [0. General bugs and questions] [answered]
### Ask a question, start a conversation, or make an annoucement

Is it possible to embed Gitbook pages into my react application via IFrame?

**Comment by SamyPesse** [comment, marked as answer]
Yes, content published **publicly** on GitBook can be embedded into an iframe. Our Content-Security-Policy are configured to allow this in a safe way (no cookies or credentials can be accessed as none is stored in public content).

Quick example with our own docs:

<img width="1332" alt="Screenshot 2024-03-07 at 10 40 53" src="https://github.com/GitbookIO/community/assets/845425/0f71d66a-f464-4777-a2f0-b2b8c3dea123">

**Reply by sheiquezakir** [reply]
I've created space in Gitbook and made it public but it was not visible to me. This is my code

<img width="989" alt="Screenshot 2024-03-07 at 3 17 57 PM" src="https://github.com/GitbookIO/community/assets/69828835/78911392-f7c4-447b-948d-1b50099f30c0">



 <iframe
                src="https://zakir-2.gitbook.io/untitled/"
                title="GitBook Embed"
                width="100%"
                height="600px"
              ></iframe>}

**Reply by SamyPesse** [reply]
It works for me, can you make sure it's not a limitation of CSP in your own application preventing the use of iframes?

<img width="1524" alt="Screenshot 2024-03-07 at 10 53 50" src="https://github.com/GitbookIO/community/assets/845425/2c9834b7-c364-41e2-8225-544f81645bbe">

**Reply by sheiquezakir** [reply]
I'm already using iframes in my application, I think there is no problem with iframs. 
<img width="1252" alt="Screenshot 2024-03-07 at 3 27 58 PM" src="https://github.com/GitbookIO/community/assets/69828835/901b830d-6ddf-45d0-b7f6-15340203e84b">

**Reply by SamyPesse** [reply]
Can you share the logs you get in the browser? Chrome should give indication of why it refuses to connect

**Reply by sheiquezakir** [reply]
I'm getting this error


Refused to frame 'https://zakir-2.gitbook.io/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors https:".

**Reply by chespinoza** [reply]
I'm having the same problem. Any way to work around this? @SamyPesse?

**Comment by chespinoza** [comment]
Alright, now that I understand what's going on, this issue occurs when running from an HTTP server without TLS, as mentioned by Samy. The headers require a secure connection, so you need to have SSL. You can get this working by generating a self-signed SSL certificate and setting up your local server to use it.

I managed to get it working using this method.

Best,

**Comment by addisonschultz** [comment]
Hi @sheiquezakir @chespinoza 

Addison here from GitBook - we’re working on a few things related to this feature request :)

Can you reach out to me at addison (@) gitbook.com with the Subject on “Embeddable GitBook”?

Would love to chat more!

**Comment by JaniceManwiller** [comment]
When embedding content, is it possible to override some of the space/site settings?

For example, our regular doc sites all have a left-hand table of contents.

However, if we were to embed the content of a topic in our application as a sort of context-sensitive help, to reduce the width and preserve the focus, we might just want the content of the topic and not the table of contents.

**Reply by addisonschultz** [reply]
Hi @JaniceManwiller - can you also reach out to me at addison (@) gitbook.com? Happy to chat with you more on this!
